src/panels/KaitoModelsPanel.ts

```diff
@@ -0,0 +1,295 @@
+import { BasePanel, PanelDataProvider } from "./BasePanel";
+import * as vscode from "vscode";
+import * as k8s from "vscode-kubernetes-tools-api";
+import { failed } from "../commands/utils/errorable";
+import { MessageHandler, MessageSink } from "../webview-contract/messaging";
+import { ToVsCodeMsgDef, ToWebViewMsgDef } from "../webview-contract/webviewDefinitions/kaitoModels";
+import { InitialState } from "../webview-contract/webviewDefinitions/kaitoModels";
+import { TelemetryDefinition } from "../webview-contract/webviewTypes";
+import { invokeKubectlCommand } from "../commands/utils/kubectl";
+import { tmpdir } from "os";
+import { writeFileSync } from "fs";
+import { join } from "path";
+// import { DeploymentState } from "../webview-contract/webviewDefinitions/periscope";
+
+export class KaitoModelsPanel extends BasePanel<"kaitoModels"> {
+    constructor(extensionUri: vscode.Uri) {
+        super(extensionUri, "kaitoModels", {
+            deploymentProgressUpdate: null,
+        });
+    }
+}
+// interface DeploymentProgressUpdate {
+//     modelName: string;
+//     workspaceExists: boolean;
+//     resourceReady: boolean | null;
+//     inferenceReady: boolean | null;
+//     workspaceReady: boolean | null;
+//     age: number | null;
+// }
+export class KaitoModelsPanelDataProvider implements PanelDataProvider<"kaitoModels"> {
+    public constructor(
+        readonly clusterName: string,
+        readonly subscriptionId: string,
+        readonly resourceGroupName: string,
+        readonly armId: string,
+        readonly kubectl: k8s.APIAvailable<k8s.KubectlV1>,
+        readonly kubeConfigFilePath: string,
+    ) {
+        this.clusterName = clusterName;
+        this.subscriptionId = subscriptionId;
+        this.resourceGroupName = resourceGroupName;
+        this.armId = armId;
+        this.kubectl = kubectl;
+        this.kubeConfigFilePath = kubeConfigFilePath;
+    }
+    private cancelToken: boolean = false;
+    getTitle(): string {
+        return `Create KAITO Workspace`;
+    }
+    getInitialState(): InitialState {
+        return {
+            clusterName: this.clusterName,
+            modelName: "",
+            workspaceExists: false,
+            resourceReady: false,
+            inferenceReady: false,
+            workspaceReady: false,
+            age: 0,
+        };
+    }
+    cancel() {
+        this.cancelToken = true;
+        // vscode.window.showInformationMessage("Operation cancelled"); - debug 1
```

Comment:
💡 I see this was used as `debug points` might be worth to get rid of all of them, but I will comment in case I see more. (as reminder) thanks :)